app/services/candidate_interface/validate_course_query_string_params.rb

```diff
@@ -0,0 +1,46 @@
+module CandidateInterface
+  class ValidateCourseQueryStringParams
+    attr_accessor :return_course
+
+    def initialize(provider_code:, course_code:)
+      @provider_code = provider_code
+      @course_code = course_code
+      @can_apply_on_apply = false
+      @course_on_find = false
+      @return_course = nil
```

Comment:
Perhaps it's easier to follow the logic in `execute` if all values are set explicitly, and we don't rely on defaults.